extensions/standard-processors/processors/SplitRecord.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "SplitRecord.h"
+
+#include "core/Resource.h"
+#include "nonstd/expected.hpp"
+#include "utils/GeneralUtils.h"
+
+namespace org::apache::nifi::minifi::processors {
+namespace {
+template<typename RecordSetIO>
+std::shared_ptr<RecordSetIO> getRecordSetIO(core::ProcessContext& context, const core::PropertyReference& property) {
+  std::string service_name;
+  if (context.getProperty(property, service_name) && !IsNullOrEmpty(service_name)) {
+    auto record_set_io = std::dynamic_pointer_cast<RecordSetIO>(context.getControllerService(service_name));
+    if (!record_set_io)
+      return nullptr;
+    return record_set_io;
+  }
+  return nullptr;
+}
+}  // namespace
+
+void SplitRecord::onSchedule(core::ProcessContext& context, core::ProcessSessionFactory&) {
+  record_set_reader_ = getRecordSetIO<core::RecordSetReader>(context, RecordReader);
+  if (!record_set_reader_) {
+    throw Exception(ExceptionType::PROCESS_SCHEDULE_EXCEPTION, "Record Reader set is missing or invalid");
+  }
+  record_set_writer_ = getRecordSetIO<core::RecordSetWriter>(context, RecordWriter);
+  if (!record_set_writer_) {
+    throw Exception(ExceptionType::PROCESS_SCHEDULE_EXCEPTION, "Record Writer set is missing or invalid");
+  }
+}
+
+nonstd::expected<std::size_t, std::string> SplitRecord::readRecordsPerSplit(core::ProcessContext& context, const core::FlowFile& original_flow_file) {
+  std::string value;
+  std::size_t records_per_split = 0;
+  if (context.getProperty(RecordsPerSplit, value, &original_flow_file)) {
+    if (!core::Property::StringToInt(value, records_per_split)) {
+      return nonstd::make_unexpected("Failed to convert Records Per Split property to an integer");
+    } else if (records_per_split < 1) {
+      return nonstd::make_unexpected("Records per split should be set to a number larger than 0");
+    }
+  } else {
+    return nonstd::make_unexpected("Records per split should be set to a valid number larger than 0");
```

Comment:
```suggestion
      return nonstd::make_unexpected("Records Per Split should be set to a number larger than 0");
    }
  } else {
    return nonstd::make_unexpected("Records Per Split should be set to a valid number larger than 0");
```

Reply:
Fixed in b34afbfa89aae6398cfdafcb017e70e13934639c